default/scripting/techs/defense/mines.macros

```diff
@@ -271,3 +190,17 @@ EG_SYSTEM_MINES
                     empire = Target.Owner
             ]
 '''
+
+SYSTEM_CONTAINS_SOURCE_OR_OWNED_PLANET
+'''                ContainedBy And [
+                    System 
+                    Contains Or [
+                        Source
```

Comment:
Source is never a special or a tech; specials are not objects. The source could be an object the special was applied to, such as a planet, or the capital of an empire in the case of a tech.